Makefile

```diff
@@ -5,7 +5,7 @@ GRAMMAR = data/cprog.gram
 TESTFILE = data/cprog.txt
 TIMEFILE = data/xxl.txt
 
-pegen/parse.c: $(GRAMMAR) pegen/*.py
+pegen/parse.c: $(GRAMMAR) pegen/*.py pegen/pegen.c pegen/*.h
 	$(PYTHON) -m pegen -q -c $(GRAMMAR) -o pegen/parse.c --compile-extension
```

Comment:
I just realized this is abusing make's dependency mechanism. The make rule on this line uses `pegen/parse.c` as the target, but the targets that depend on it (`dump`, `test`, `time`) all need the *side effect* of using `--compile-extension`, which produces a `.so` file.

On my Mac that `.so` file is called `pegen/parse.cpython-38-darwin.so`. If it's also called that on Linux, let's name that file instead of `pegen/parse.c` as the target here and as the dependency of the other targets.

If it's got a different name on Linux, let's keep it this way, but let's add a comment.

Reply:
Yeah, you guessed the reason for this "hack" and why I used the implicit dependency on `--compile-extension`: on Linux this is called `parse.cpython-38-x86_64-linux-gnu.so` (and this can change in other platforms). I will add a comment to explain that.